services/data/src/links/RestAPILink/fetchData.ts

```diff
@@ -75,12 +74,20 @@ export function fetchData(
         })
         .then(parseStatus)
         .then(async response => {
-            if (
-                parseContentType(response.headers.get('Content-Type')) ===
-                'application/json'
-            ) {
+            const contentType = parseContentType(
+                response.headers.get('Content-Type')
+            )
+
+            // 'application/json'
+            if (contentType === 'application/json') {
                 return await response.json() // Will throw if invalid JSON!
             }
-            return await response.text()
+
+            // 'text/*'
+            if (/^text\/[a-z0-9.-]*$/.test(contentType)) {
+                return await response.text()
+            }
+
+            return await response.blob()
```

Comment:
This makes sense to me, however I worry a bit about the consumer again... if the API changes the response type, the consumer code will break...  not sure we can do anything about that here, though

Reply:
(from Slack) let's keep this for now, it's an improvement on the existing behavior which assumes text.